backend/src/interfaces.ts

```diff
@@ -0,0 +1,36 @@
+export enum GameState {
+  IN_PROGRESS = 'in_progress',
+  FINISHED = 'finished'
+}
+
+// for destination, I'm assuming it will be a link
+// I'll change it if needed
+export interface ButtonCoordinates {
+  x: number;
+  y: number;
+  destination: string;
+}
+
+export interface Level {
```

Comment:
this looks good, consider adding a reference to the level in the database - i believe the type for this would be `firebase.firestore.DocumentSnapshot.id`. and maybe just call it `id`. you'll probably need to import firestore etc into this file to reference this

Reply:
i'm having trouble with the `firebase.firestore.DocumentSnapshot.id`. I imported DocumentSnapshot using `import { DocumentSnapshot } from "firebase/firestore";` but if i do `id: DocumentSnapshot.id` it gives an error saying `Cannot access 'DocumentSnapshot.id' because 'DocumentSnapshot' is a type, but not a namespace.` i also tried firebase.firestore.DocumentSnapshot.id but did not work.

Reply:
thanks for giving it ago, just use string, that'll be fine for our purposes

Reply:
great step in the right direction, i'd just recommend changing xcoord/ycoord to 'latitude' and 'longitude'. 'zCoordinate' being 'zPosition' might make a little more sense as well? floor 3 in a building is not really a coordinate if that makes sense

Reply:
done 👍